assets/app/scripts/directives/logViewer.js

```diff
@@ -69,46 +84,39 @@ angular.module('openshiftConsole')
                 });
               }
             };
-            $(window).scroll(onScroll);
+            $win.scroll(onScroll);
+
 
             var onResize = _.debounce(updateScrollLinks, 50);
-            $(window).resize(onResize);
+            $win.on('resize', onResize);
 
-            var scrollBottom = function() {
+
+            var autoScrollBottom = function() {
               // Tell the scroll listener this is an auto-scroll. The listener
               // will reset it to false.
               autoScrollingNow = true;
-              logLinks.scrollBottom();
+              logLinks.scrollBottom(scrollableDOMNode);
             };
 
             var toggleAutoScroll = function() {
               $scope.autoScroll = !$scope.autoScroll;
               if ($scope.autoScroll) {
                 // Scroll immediately. Don't wait the next message.
-                scrollBottom();
+                autoScrollBottom();
               }
             };
 
-            var scrollTop = function() {
-              // Stop auto-scrolling when the user clicks the scroll top link.
-              $scope.autoScroll = false;
-              logLinks.scrollTop();
-            };
 
-            var buffer = document.createDocumentFragment();
+            var buffer = clearBuffer();
```

Comment:
The function is not really clearing a buffer. It's creating a doc fragment. I feel like pulling this one-liner into a function is making it harder to follow.